utils/tx.go

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"crypto/sha256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetTxHash returns the hash of the current transaction
+func GetTxHash(ctx sdk.Context) []byte {
+	tx := ctx.TxBytes()
+	hash := sha256.Sum256(tx)
+	return hash[:]
+}
+
+// GetTxHashAsHex returns the hash of the current transaction as a hex string
+func GetTxHashAsHex(ctx sdk.Context) string {
```

Comment:
we have a wrapper Hash type in EVM module but prefer to avoid refactoring that in this PR